tests/integration/system/apiauth/api_auth_test.go

```diff
@@ -0,0 +1,226 @@
+	suite.adminClient = testutils.GetHTTPClient()
+	suite.plainClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	suite.invalidTokenClient = testutils.GetHTTPClientWithToken("invalid-token")
+
+	ouID, err := testutils.CreateOrganizationUnit(testutils.OrganizationUnit{
+		Handle: fmt.Sprintf("api-auth-ou-%d", time.Now().UnixNano()),
+		Name:   "API Auth Test OU",
+	})
+	suite.Require().NoError(err)
+	suite.ouID = ouID
+
+	userSchema := testutils.UserSchema{
+		Name:                  fmt.Sprintf("api-auth-user-%d", time.Now().UnixNano()),
+		OrganizationUnitId:    suite.ouID,
+		AllowSelfRegistration: true,
+		Schema: map[string]interface{}{
+			"username":  map[string]interface{}{"type": "string"},
+			"password":  map[string]interface{}{"type": "string"},
+			"email":     map[string]interface{}{"type": "string"},
+			"firstName": map[string]interface{}{"type": "string"},
+			"lastName":  map[string]interface{}{"type": "string"},
+		},
+	}
+
+	userSchemaID, err := testutils.CreateUserType(userSchema)
+	suite.Require().NoError(err)
+	suite.userSchemaID = userSchemaID
+
+	username := fmt.Sprintf("authuser_%d", time.Now().UnixNano())
+	password := "ApiAuthTest123!"
+	userAttrs := map[string]interface{}{
+		"username":  username,
+		"password":  password,
+		"email":     fmt.Sprintf("%s@example.com", username),
+		"firstName": "API",
+		"lastName":  "Auth",
+	}
+	attrBytes, err := json.Marshal(userAttrs)
+	suite.Require().NoError(err)
+
+	userID, err := testutils.CreateUser(testutils.User{
+		OrganizationUnit: suite.ouID,
+		Type:             userSchema.Name,
+		Attributes:       attrBytes,
+	})
+	suite.Require().NoError(err)
+	suite.regularUserID = userID
+
+	userClient, err := testutils.GetHTTPClientForUser(username, password)
+	suite.Require().NoError(err)
+	suite.userClient = userClient
+}
+
+func (suite *APIAuthTestSuite) TearDownSuite() {
+	if suite.regularUserID != "" {
+		if err := testutils.DeleteUser(suite.regularUserID); err != nil {
+			suite.T().Logf("Failed to delete regular user: %v", err)
+		}
+	}
+
+	if suite.userSchemaID != "" {
+		if err := testutils.DeleteUserType(suite.userSchemaID); err != nil {
+			suite.T().Logf("Failed to delete user schema: %v", err)
+		}
+	}
+
+	if suite.ouID != "" {
+		if err := testutils.DeleteOrganizationUnit(suite.ouID); err != nil {
+			suite.T().Logf("Failed to delete organization unit: %v", err)
+		}
+	}
+}
+
+// Authentication: valid system token succeeds.
+func (suite *APIAuthTestSuite) TestSystemTokenAuthorized() {
+	req, err := http.NewRequest(http.MethodGet, suite.protectedResourceURL(), nil)
+	suite.Require().NoError(err)
+
+	resp, err := suite.adminClient.Do(req)
+	suite.Require().NoError(err)
+	defer closeBodyQuietly(suite.T(), resp.Body)
+
+	suite.Equal(http.StatusOK, resp.StatusCode)
+
+	var ou testutils.OrganizationUnit
+	suite.Require().NoError(json.NewDecoder(resp.Body).Decode(&ou))
+	suite.Equal(suite.ouID, ou.ID)
```

Comment:
This will validate that the response body aligns with the existing OU struct. This can be broken when a attribute is removed from the thunder API side. 
We need this test. Shouldn't this be covered in some other test? WDYT?